packages/website/docs/creating-multi-source-autocompletes.md

```diff
@@ -2,3 +2,375 @@
+
+In this example, [`getItems`](sources/#getitems) returns a filtered array of `predefinedItems`. The code filters the array to return items that match the query, if it exists. If it doesn't, it returns the entire array. **You can return whatever predefined items you like and format them accordingly.** For example, suppose you want to show trending search items instead of helpful links. Then, you can use [`getItems`](sources/#getitems) to retrieve them from another source, [including an asynchronous API](sources#using-dynamic-sources).
+
+The [`getItemUrl`](sources/#getitemurl) function defines how to get the URL of an item. In this case, since it's an attribute on each object in the `predefinedItems` array, you can simply return the attribute. You can use [`getItemUrl`](sources/#getitemurl) to add [keyboard navigation](keyboard-navigation) to the autocomplete menu. Users can scroll through items in the autocomplete menu with the arrow up and down keys. When they hit enter on one of the `predefinedItems` (or any source that includes[`getItemUrl`](sources/#getitemurl)), it opens the URL retrieved from [`getItemUrl`](sources/#getitemurl).
+
+[Templates](templates) define how to display each section of the autocomplete, including the [`header`](templates#header), [`footer`](templates#footer), and each [`item`](templates#item). You can provide either a string, or as the example below shows, a function for how to manipulate the DOM.
+
+```diff
+  const predefinedItems = [
+  {
+    label: 'Documentation',
+    url: 'https://autocomplete.algolia.com/',
+  },
+  {
+    label: 'GitHub',
+    url: 'https://github.com/algolia/autocomplete.js/tree/next',
+  },
+]
+export const predefinedItemsPlugin = {
+  getSources() {
+    return [
+      {
+        getItems({ query }) {
+          return predefinedItems.filter(
+            (item) =>
+              !query || item.label.toLowerCase().includes(query.toLowerCase())
+          );
+        },
+        getItemUrl({ item }) {
+          return item.url;
+        },
+        templates: {
++          item({ item}) {
++              return (
++                <a href={item.url} className="aa-ItemContent aa-ItemLink aa-PredefinedItem">
++                  <div className="aa-ItemSourceIcon">
++                    <svg style="width:20px;height:20px" viewBox="0 0 24 24">
++                       <path fill="currentColor" d="M14,3V5H17.59L7.76,14.83L9.17,16.24L19,6.41V10H21V3M19,19H5V5H12V3H5C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V12H19V19Z" />
++                     </svg>
++                  </div>
++                  <div className="'aa-ItemTitle">{item.label}</div>
++                </a>
++              )
++        },
+        },
+      },
+    ];
+  },
+};
+```
+
+### Adding CSS for the predefined items
+
+To style the predefined links, create a `predefinedItemsPlugin.css` file in your `src` directory, and add this CSS to it:
+
+```css
+.aa-ItemLink {
```

Comment:
This isn't required with the new theme now.

We need to update the markup of this guide to the new one: https://github.com/algolia/autocomplete.js/blob/8a4da0211d9f671efe1c26f1ec6179fb4a77ff1c/examples/js/app.tsx#L112-L137

Reply:
This markup is for a product from the instant_search index, right? Since the predefined items don't have the `image`,`name`, and `description` attributes I'm not sure this is the right markup? or did I misunderstand?

Reply:
In the new theme we already design `.aa-ItemLink` so there's no need for users to override it.

Don't sweat too much about the markup for now indeed, I'll do a final pass.